code/common/datastripe.q

```diff
@@ -17,3 +17,28 @@ getendtime:{[x] max x[`time]};
 
 // function to clear tables before given time
 deletetablebefore:{![x;enlist (<;y;z);0b;0#`]}
+
+filterfunc:{[lf;td;logmetatab]
+  // lf is a log file handle and td is a dictionary with table names as keys and where clauses to filter by as values
+  // logmetatab is the tplog metadata table loaded into the stp from the tplogs directory
+    .lg.o[`subscribe;"replaying log file ",.Q.s1 lf]; -11!lf;
+  // checks if the log file contains a table that requires filtering
+    filtertab:(key td) inter  raze (select from logmetatab where logname=@[lf;1])`tbls;
+  // filters tables replayed by the logs if required
+    if[count filtertab;applyfilters[filtertab;td]];
+    };
+
+// function to apply datastriping filter from a filter dictionary to a table
+tablesfilter:{[filtertab;td]
```

Comment:
nitpick on name - this function only filters one table, not "tables". imo `filtertable` would be better